src/test/java/com/unit/UNIT/TEST/MachineTest.java

```diff
@@ -109,4 +110,51 @@ public void shouldNotIncreaseAmountOfMoney() {
         assertEquals(finalResult, machine.bankAmount);
         verify(validator,times(1)).validateMoney(money);
     }
+
+    @Test
+    public void shouldGenerateThreeGameSigns() {
+        //given
+        int numberOfGameSigns = 3;
+        //when
+        machine.generateThreeSigns();
+        //then
+        assertEquals(numberOfGameSigns, machine.gameSigns.size());
+    }
+
+    @Test
+    public void shouldGenerateGameSignsInRangeFromZeroToNine() {
+        //given
+        int lowerBound = 0;
+        int upperBound = 9;
+        //when
+        machine.generateThreeSigns();
+        //then
+        for(int gameSign : machine.gameSigns) {
+            assertTrue(gameSign >= lowerBound);
+            assertTrue(gameSign <= upperBound);
+        }
+    }
+
+    @Test
+    public void shouldNotStartGameWhenInsertedInvalidMoney() {
+        //given
+        Machine machineSpy = Mockito.spy(machine);
```

Comment:
Testujesz Machine, a wykorzystujesz spy tylko po to, żeby zrobić verify. Pomysł co najmniej ciekawy. 👍 Jednak tutaj bym się pokusił o testowanie logiki aniżeli ilości wywołań.